script/demoScripts/demoGasZip.ts

```diff
@@ -0,0 +1,87 @@
+import { parseUnits, Narrow, zeroAddress } from 'viem'
+import { randomBytes } from 'crypto'
+import dotenv from 'dotenv'
+import gasZipFacetArtifact from '../../out/GasZipFacet.sol/GasZipFacet.json'
+import { ILiFi } from '../../typechain'
+import { SupportedChain } from './utils/demoScriptChainConfig'
+import {
+  ensureBalance,
+  executeTransaction,
+  setupEnvironment,
+} from './utils/demoScriptHelpers'
+import { IGasZip } from '../../typechain/GasZipFacet'
+import { PublicKey } from '@solana/web3.js'
+
+dotenv.config()
+
+// #region ABIs
+const GAS_ZIP__FACET_ABI = gasZipFacetArtifact.abi as Narrow<
+  typeof gasZipFacetArtifact.abi
+>
+// #endregion
+
+const NON_EVM_ADDRESS = '0x11f111f111f111F111f111f111F111f111f111F1'
+
+async function main() {
+  // === Set up environment ===
+  const srcChain: SupportedChain = 'arbitrum'
+  const destinationChainId = 245 // solana (non evm) -  custom destination chain id for gas.zip - check here (https://dev.gas.zip/gas/chain-support/outbound)
+
+  const { publicClient, walletAccount, lifiDiamondContract } =
+    await setupEnvironment(srcChain, GAS_ZIP__FACET_ABI)
+  const signerAddress = walletAccount.address
+  // const userReceiver = addressToBytes32RightPadded({evm_address}) // <== in case of evm address
+  const userReceiver = `0x${new PublicKey(
+    'DDMe5C8EhVhaVZRu3ukqhXF5CqnjuxhxbXBXj7pZnTw6'
+  )
+    .toBuffer()
+    .toString('hex')}`
+
+  // === Contract addresses ===
+  const SRC_TOKEN_ADDRESS = zeroAddress as `0x${string}` // native token
+
+  const amount = parseUnits('0.001', 18) // 0.001 * 1e18
+
+  console.info(`Bridge ${amount} native from ${srcChain} --> Solana`)
+  console.info(`Connected wallet address: ${signerAddress}`)
+
+  await ensureBalance(SRC_TOKEN_ADDRESS, signerAddress, amount, publicClient)
+
+  // === Prepare bridge data ===
+  const bridgeData: ILiFi.BridgeDataStruct = {
+    transactionId: `0x${randomBytes(32).toString('hex')}`,
+    bridge: 'gasZip',
+    integrator: 'ACME Devs',
+    referrer: zeroAddress,
+    sendingAssetId: zeroAddress, // <-- native token
+    receiver: NON_EVM_ADDRESS,
+    destinationChainId,
```

Comment:
while it's not a problem for the script itself, I just want to point out that you are using GasZipChainId as "normal" chainId here which would be wrong if you were developing in the BE
The correct chainId used here should be 1151111081099710 which is our own custom chainId for Solana.

Reply:
hmm ok, looking at for example RelayFacet then inside of the GasZipFacet we should have mapping from our BE chains to GasZipChainId mapping. Am I right?

Is this BE mapping somewhere available?

Reply:
No we do not need a mapping. 
`GasZipData` in `IGasZip.sol` has a `destinationChains` parameter which is used to signal to GasZip where to send the funds to (it can send to many chains with one call, so all chainIds are compressed into one value). 

Which actually makes me realize that your demoScript is wrong. 
You need to construct that destinationChains parameter correctly in your demo script as described here: 

https://dev.gas.zip/gas/code-examples/evm-deposit/contract-deposit

Reply:
I see that we already have `getDestinationChainsValue` function that is already in `GasZipFacet` and `GasZipPeriphery` and in both contracts this function is not used. It accepts `uint8[] calldata _chainIds` and returns `uint256 destinationChains`. GasZipData has `uint256 destinationChains` so I assume in periphery contract constructing dectinationChains parameter is already done on the backend side.

Do you know more context. If it works this way then we can remove `getDestinationChainsValue` function.

Reply:
This function is a convenience function for third parties to have an easy way to calculate that parameter. 
I do believe it makes sense to keep it there.

Reply:
you could use that function in your demo script by the way....